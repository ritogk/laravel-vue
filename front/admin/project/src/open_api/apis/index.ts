/* tslint:disable */
/* eslint-disable */
export * from './AuthAdminApi';
export * from './AuthFrontApi';
export * from './EntryApi';
export * from './FileApi';
export * from './JobApi';
export * from './JobCategorieApi';
export * from './UserApi';
