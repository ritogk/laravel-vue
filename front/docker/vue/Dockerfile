FROM node:14.19.0-buster

WORKDIR /usr/src/app

RUN apt-get update \
&& apt-get install -y \
&& npm install \