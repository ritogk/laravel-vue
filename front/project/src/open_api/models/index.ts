/* tslint:disable */
/* eslint-disable */
export * from './AccessToken';
export * from './Admin';
export * from './Entry';
export * from './Job';
export * from './JobCategory';
export * from './ModelFile';
export * from './QueryEntryList';
export * from './QueryJobCategoryList';
export * from './QueryJobList';
export * from './QueryUserList';
export * from './RequestEntry';
export * from './RequestFile';
export * from './RequestJob';
export * from './RequestJobCategory';
export * from './RequestLogin';
export * from './RequestUser';
export * from './Sex';
export * from './User';
