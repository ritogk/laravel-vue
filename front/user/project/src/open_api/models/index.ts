/* tslint:disable */
/* eslint-disable */
export * from './AccessToken';
export * from './Admin';
export * from './Entry';
export * from './FilePath';
export * from './Job';
export * from './JobCategory';
export * from './QueryEntryList';
export * from './QueryJobCategoryList';
export * from './QueryJobList';
export * from './QueryUserList';
export * from './RequestEntry';
export * from './RequestJob';
export * from './RequestJobCategory';
export * from './RequestLogin';
export * from './RequestUser';
export * from './ResponseJobEntry';
export * from './Sex';
export * from './User';
