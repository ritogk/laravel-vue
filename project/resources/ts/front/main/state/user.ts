import { Mutation, Action, VuexModule, getModule, Module } from "vuex-module-decorators";
import Vuex from 'vuex';
import Vue from 'vue';

import IUser from "@root/models/IUser";

// state's interface
export interface IState {
    loginUser: IUser
}

// ログイン会員情報 初期値
const dafaultLoginUser: IUser = {
    user:{id: '', name: '',email: '', email_verified_at: '', self_pr: '', tel: '', created_at: '', updated_at: ''},
    status: false
};

Vue.use(Vuex);
const store=new Vuex.Store({})

@Module({ dynamic: true, store, name: "state", namespaced: true })
class State extends VuexModule implements IState {
    // state
    // ログイン会員情報
    loginUser: IUser = dafaultLoginUser;

    // getter
    public get getLoginUser(): IUser{
        return this.loginUser
    }

    // mutation
    @Mutation
    public setLoginUser(value: IUser) {
        this.loginUser = value;
    }

    // action
    @Action
    // ログイン会員情報の初期化
    public init() {
        this.setLoginUser(dafaultLoginUser)
    }
}
export const state = getModule(State);
