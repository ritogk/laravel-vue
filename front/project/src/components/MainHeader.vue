<template>
  <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
    <div class="container-fluid">
      <a class="navbar-brand" href="#" @click="clickTitle">求人検索</a>
      <button
        class="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse text-end" id="navbarNav">
        <ul class="navbar-nav ms-auto">
          <li class="nav-item">
            <a class="nav-link active" aria-current="page" href="#"
              >ログアウト</a
            >
          </li>
          <li class="nav-item">
            <a class="nav-link" href="#">ユーザー名</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="#" @click="clickLogin">ログイン</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="#" @click="clickRegistration">新規登録</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="#">管理画面</a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import { useRouter } from 'vue-router';

export default defineComponent({
  setup() {
    const router = useRouter();
    // 「タイトル」押下
    const clickTitle = () => {
      router.push('/');
    };
    // 「新規登録」押下
    const clickRegistration = () => {
      router.push({ name: 'UserRegistration' });
    };
    // 「ログイン」押下
    const clickLogin = () => {
      router.push({ name: 'UserLogin' });
    };
    return { clickTitle, clickRegistration, clickLogin };
  },
});
</script>
