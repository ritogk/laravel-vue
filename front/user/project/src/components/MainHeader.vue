<template>
  <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
    <div class="container-fluid">
      <a class="navbar-brand" href="#" @click="clickTitle">求人検索</a>
      <button
        class="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse text-end" id="navbarNav">
        <ul class="navbar-nav ms-auto">
          <li class="nav-item">
            <!-- ログアウト-->
            <a
              class="nav-link active"
              aria-current="page"
              href="#"
              @click="clickLogout"
              v-show="isLogin"
              >ログアウト</a
            >
          </li>
          <!-- ログイン済のユーザーの氏名-->
          <li class="nav-item">
            <a
              class="nav-link"
              href="#"
              v-text="loginUser.name"
              v-show="isLogin"
            ></a>
          </li>
          <!-- ログイン -->
          <li class="nav-item">
            <a class="nav-link" href="#" @click="clickLogin" v-show="!isLogin"
              >ログイン</a
            >
          </li>
          <!-- 新規登録 -->
          <li class="nav-item">
            <a
              class="nav-link"
              href="#"
              @click="clickRegistration"
              v-show="!isLogin"
              >新規登録</a
            >
          </li>
        </ul>
      </div>
    </div>
  </nav>
</template>

<script lang="ts">
import { defineComponent, inject } from 'vue';
import { useRouter } from 'vue-router';
import {
  userAuthenticationKey,
  useUserAuthenticationType,
} from '@/composables/useUserAuthentication';

export default defineComponent({
  setup() {
    const router = useRouter();
    const { userAuthenticationRefs, logout } = inject(
      userAuthenticationKey
    ) as useUserAuthenticationType;

    // 「タイトル」押下
    const clickTitle = () => {
      router.push('/');
    };
    // 「新規登録」押下
    const clickRegistration = () => {
      router.push({ name: 'UserRegistration' });
    };
    // 「ログアウト」押下
    const clickLogout = () => {
      logout();
    };
    // 「ログイン」押下
    const clickLogin = () => {
      router.push({ name: 'UserLogin' });
    };
    return {
      loginUser: userAuthenticationRefs.user,
      isLogin: userAuthenticationRefs.isLogin,
      clickTitle,
      clickRegistration,
      clickLogin,
      clickLogout,
    };
  },
});
</script>
