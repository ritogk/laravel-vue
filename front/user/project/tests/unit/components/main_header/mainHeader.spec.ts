import { nextTick } from 'vue';
import { flushPromises, mount, VueWrapper } from '@vue/test-utils';
import * as vueRouter from 'vue-router';
import {
  useUserAuthentication,
  userAuthenticationKey,
} from '@/composables/useUserAuthentication';
import { User } from '@/open_api';
import { userModel } from '~/utils/data/user';
// テスト対象のコンポーネント
import MainHeader from '@/components/MainHeader.vue';

describe('未ログイン状態での「MainHeader.vue」コンポーネントのテスト', () => {
  let wrapper: VueWrapper<any>;

  /**
   * 外部からの振る舞いを設定
   */
  beforeEach(async () => {
    // モックを作らないとなぜかエラーがでる
    jest
      .spyOn(vueRouter, 'useRouter')
      .mockImplementation((): any => ({ push: () => {} }));

    // 未ログイン状態のモックを作成する
    const objUseUserAuthentication = useUserAuthentication();
    objUseUserAuthentication.userAuthenticationRefs.isLogin.value = false;
    objUseUserAuthentication.userAuthenticationRefs.user.value = {} as User;

    // コンポーネントをレンダリング
    wrapper = mount(MainHeader, {
      global: {
        provide: {
          [userAuthenticationKey as symbol]: objUseUserAuthentication,
        },
      },
    });

    await flushPromises();
    await nextTick();
  });

  /**
   * 確認項目
   */
  it('スナップショットとの差分を確認する', async () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('ログイン済状態での「MainHeader.vue」コンポーネントのテスト', () => {
  let wrapper: VueWrapper<any>;

  /**
   * 外部からの振る舞いを設定
   */
  beforeEach(async () => {
    // モックを作らないとなぜかエラーがでる
    jest
      .spyOn(vueRouter, 'useRouter')
      .mockImplementation((): any => ({ push: () => {} }));

    // ログイン済の状態のモックを作成する
    const objUseUserAuthentication = useUserAuthentication();
    objUseUserAuthentication.userAuthenticationRefs.isLogin.value = true;
    objUseUserAuthentication.userAuthenticationRefs.user.value = userModel;

    // コンポーネントをレンダリング
    wrapper = mount(MainHeader, {
      global: {
        provide: {
          [userAuthenticationKey as symbol]: objUseUserAuthentication,
        },
      },
    });

    await flushPromises();
    await nextTick();
  });

  /**
   * 確認項目
   */
  it('スナップショットとの差分を確認する', async () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
