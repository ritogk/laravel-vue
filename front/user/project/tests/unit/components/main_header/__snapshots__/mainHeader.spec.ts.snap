// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ログイン済状態での「MainHeader.vue」コンポーネントのテスト スナップショットとの差分を確認する 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <div
    class="container-fluid"
  >
    <a
      class="navbar-brand"
      href="#"
    >
      求人検索
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse text-end"
      id="navbarNav"
    >
      <ul
        class="navbar-nav ms-auto"
      >
        <li
          class="nav-item"
        >
          <!-- ログアウト-->
          <a
            aria-current="page"
            class="nav-link active"
            href="#"
          >
            ログアウト
          </a>
        </li>
        <!-- ログイン済のユーザーの氏名-->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            山田太郎
          </a>
        </li>
        <!-- ログイン -->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
            style="display: none;"
          >
            ログイン
          </a>
        </li>
        <!-- 新規登録 -->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
            style="display: none;"
          >
            新規登録
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`未ログイン状態での「MainHeader.vue」コンポーネントのテスト スナップショットとの差分を確認する 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <div
    class="container-fluid"
  >
    <a
      class="navbar-brand"
      href="#"
    >
      求人検索
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse text-end"
      id="navbarNav"
    >
      <ul
        class="navbar-nav ms-auto"
      >
        <li
          class="nav-item"
        >
          <!-- ログアウト-->
          <a
            aria-current="page"
            class="nav-link active"
            href="#"
            style="display: none;"
          >
            ログアウト
          </a>
        </li>
        <!-- ログイン済のユーザーの氏名-->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
            style="display: none;"
          />
        </li>
        <!-- ログイン -->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            ログイン
          </a>
        </li>
        <!-- 新規登録 -->
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            新規登録
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
